ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/sendtransaction/TransactionTransmitter.java

```diff
@@ -86,21 +91,23 @@ private void createSignedTransactionBody() {
       } else {
         routingContext.fail(GATEWAY_TIMEOUT.code(), new JsonRpcException(INTERNAL_ERROR));
       }
-      return;
-    } catch (final Throwable thrown) {
-      LOG.debug("Failed to encode/serialize transaction: {}", transaction, thrown);
-      routingContext.fail(BAD_REQUEST.code(), new JsonRpcException(INTERNAL_ERROR));
-      return;
+      return Optional.empty();
     }
 
-    final JsonRpcRequest rawTransaction =
-        transaction.jsonRpcRequest(signedTransactionHexString, transaction.getId());
+    final String signedTransactionHexString;
     try {
-      sendTransaction(Json.encodeToBuffer(rawTransaction));
+      signedTransactionHexString = transactionSerializer.serialize(transaction);
     } catch (final IllegalArgumentException e) {
-      LOG.debug("JSON Serialization failed for: {}", rawTransaction, e);
+      LOG.debug("Failed to encode transaction: {}", transaction, e);
```

Comment:
Since the error message being logged is the same for this and below exception clause, perhaps just remove this exception clause and use the generic Throwable below?

Reply:
but the JsonRpcError reported is different - one is definitively "INVALID PARAMS" . -the one under "Throwable" is simply - something real bad happened.